src/test/java/apoc/es/ElasticSearchTest.java

```diff
@@ -11,42 +14,272 @@
+import java.util.UUID;
 
-import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.*;
 
 /**
  * @author mh
  * @since 21.05.16
  */
 public class ElasticSearchTest {
 
-    private final static String HOST = "localhost";
     private final static String ES_INDEX = "test-index";
+
     private final static String ES_TYPE = "test-type";
-    private final static String ES_ID = "1";
 
-    private static GraphDatabaseService db;
+    private final static String ES_ID = UUID.randomUUID().toString();
 
+    private final static String HOST = "localhost";
+
+    protected static GraphDatabaseService db;
     // We need a reference to the class implementing the procedures
     private final ElasticSearch es = new ElasticSearch();
+    private Configuration JSON_PATH_CONFIG = Configuration.builder().options(Option.DEFAULT_PATH_LEAF_TO_NULL, Option.SUPPRESS_EXCEPTIONS).build();
 
     @BeforeClass
     public static void setUp() throws Exception {
         db = new TestGraphDatabaseFactory()
                 .newImpermanentDatabaseBuilder()
                 .newGraphDatabase();
         TestUtil.registerProcedure(db, ElasticSearch.class);
+        TestUtil.registerProcedure(db, Periodic.class);
+
+        insertDocuments();
     }
 
     @AfterClass
     public static void tearDown() {
         db.shutdown();
     }
 
+    private static void insertDocuments() {
+        TestUtil.ignoreException(() -> {
+            String payload = "{procedurePackage:'es',procedureName:'get',procedureDescription:'performa a GET operation to ElasticSearch'}";
+            String id = UUID.randomUUID().toString();
+
+            TestUtil.testCall(db, "CALL apoc.es.post('" + HOST + "','" + ES_INDEX + "','" + ES_TYPE + "','" + id + "',null," + payload + ") yield value", r -> assertNotNull(r.get("value")));
+        }, ConnectException.class);
+
+        TestUtil.ignoreException(() -> {
+            String payload = "{procedurePackage:'es',procedureName:'post',procedureDescription:'performa a POST operation to ElasticSearch'}";
+            String id = UUID.randomUUID().toString();
+
+            TestUtil.testCall(db, "CALL apoc.es.post('" + HOST + "','" + ES_INDEX + "','" + ES_TYPE + "','" + id + "',null," + payload + ") yield value", r -> assertNotNull(r.get("value")));
+        }, ConnectException.class);
+
+        TestUtil.ignoreException(() -> {
+            String payload = "{name:'Neo4j',company:'Neo Technology',description:'Awesome stuff with a graph database'}";
+
+            TestUtil.testCall(db, "CALL apoc.es.post('" + HOST + "','" + ES_INDEX + "','" + ES_TYPE + "','" + ES_ID + "',null," + payload + ") yield value", r -> assertNotNull(r.get("value")));
+        }, ConnectException.class);
+    }
+
+    private final Object extractValueFromResponse(Map response, String jsonPath) {
+        Object jsonResponse = response.get("value");
+        assertNotNull(jsonResponse);
+
+        String json = JsonPath.parse(jsonResponse).jsonString();
```

Comment:
probably something for Util (later) 
I hoped that there is something in JsonPath that allows you to use the path directly on a Map or List or Object
so you don't have to go map -> string -> parse